frontend/catalyst/tracing/type_signatures.py

```diff
@@ -156,6 +156,41 @@ def split_static_args(args, static_argnums):
     return tuple(dynamic_args), tuple(static_args)
 
 
+def merge_static_argname_into_argnum(fn: Callable, static_argnames, static_argnums):
+    """Map static_argnames of the callable to the corresponding argument indices,
+    and add them to static_argnums"""
+    new_static_argnums = [] if (static_argnums is None) else list(static_argnums)
+    fn_argnames = list(inspect.signature(fn).parameters.keys())
+
+    # static_argnames can be a single str, or a list/tuple of strs
+    # convert all of them to list
+    if isinstance(static_argnames, str):
+        static_argnames = [static_argnames]
+
+    non_existent_args = []
+    for static_argname in static_argnames:
+        if static_argname in fn_argnames:
+            new_static_argnums.append(fn_argnames.index(static_argname))
+            continue
+        non_existent_args.append(static_argname)
+
+    if non_existent_args:
+        non_existent_args_str = "{"
+        for arg in non_existent_args:
+            non_existent_args_str += "'" + arg + "', "
+        non_existent_args_str = non_existent_args_str[:-2] + "}"
```

Comment:
Just so I understand, is the purpose of this loop over `non_existent_args` just to preserve the order of arguments as they appear in `static_argnames`, since `set(non_existent_args)` does not preserve that order?

Reply:
Yes; I tried various things but none seemed to work, so I have to resort to the raw way of doing it

Reply:
`set(sorted(set(non_existent_args)))` also does not preserve the order

Reply:
I do think `sorted(set(non_existent_args))` will preserve the order, but the outmost set becomes unordered again (since it's a set).

Really, we just want the braces in the error msg, so we can just build the string ourselves if making sets happy is too difficult.

Reply:
Yeah I think this is fine. Here's a one-liner that essentially does the same thing. Feel free to use it if you like.

```python
'{' + ', '.join(repr(item) for item in non_existent_args) + '}'
```

For example:

```pycon
>>> non_existent_args = ['b', 'a', 'c', '[d]', 'e']
>>> result = '{' + ', '.join(repr(item) for item in non_existent_args) + '}'
>>> print(result)
{'b', 'a', 'c', '[d]', 'e'}
```

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1158/commits/771f2fbd9a0608a7779159a636c096a05f78a015

Reply:
[Don't fear the `repr`!](https://en.wikipedia.org/wiki/(Don%27t_Fear)_The_Reaper)